desktop/core/src/desktop/management/commands/runruff.py

```diff
@@ -0,0 +1,142 @@
+
+    Options:
+      --diff-branch=<branch>  Compare the current branch with the specified branch
+                              and check only the files that have changed.
+      All original Ruff commands are supported, including --fix to automatically fix errors.
+
+    Examples:
+      Run Ruff linting on specific Python files:
+        ./build/env/bin/hue runruff check file1.py file2.py
+
+      Run Ruff linting on files changed between the current branch and a specified diff branch:
+        ./build/env/bin/hue runruff check --diff-branch=origin/master
+
+      Automatically fix all Ruff errors on files changed between the current branch and a specified diff branch:
+        ./build/env/bin/hue runruff check --diff-branch=origin/master --fix
+
+    Note:
+      Make sure to install Ruff first by running `pip install ruff` in your environment.
+      This command passes all additional arguments to Ruff, so you can use any Ruff option or flag.
+  """)
+
+  def __init__(self):
+    super().__init__()
+
+  def diff_files(self, diff_branch):
+    """
+    Returns a list of files that have changed between the current branch and the specified diff_branch.
+
+    Args:
+      diff_branch (str): The branch to compare with.
+
+    Returns:
+      list: A list of files that have changed.
+    """
+    git_cmd = ["git", "diff", "--name-only", diff_branch, "--diff-filter=bd"]
+    egrep_cmd = ["egrep", ".py$"]
+
+    try:
+      # Run the git command first and capture it's output
+      git_process = subprocess.run(git_cmd, stdout=subprocess.PIPE, check=True)
+      git_output, _ = git_process.stdout, git_process.stderr
+
+      # Next, run the egrep command on git output and capture final output to return as list of files
+      egrep_process = subprocess.run(egrep_cmd, input=git_output, stdout=subprocess.PIPE, check=True)
+      egrep_output, _ = egrep_process.stdout, egrep_process.stderr
+
+      # Decode the egrep output, strip trailing newline, split into a list of files, and return it
+      # Otherwise, return an empty list
+      return egrep_output.decode().strip('\n').split('\n') if egrep_output else []
+
+    except subprocess.CalledProcessError as e:
+      LOG.error(f"Error running git or egrep command: {e}")
+      return []
+
+  def add_arguments(self, parser):
+    parser.add_argument('ruff_args', nargs=argparse.REMAINDER, help='Additional Ruff arguments, e.g. check, format, --fix etc.')
+    parser.add_argument(
+      '--diff-branch',
+      action='store',
+      dest='diff_branch',
+      default='origin/master',
+      help='Compare with this branch to check only changed files',
+    )
+
+  def handle(self, *args, **options):
+    """
+    Handles the command.
+
+    Args:
+      *args: Variable arguments.
+      **options: Keyword arguments.
+    """
+    ruff_package = paths.get_build_dir('env', 'bin', 'ruff')
+
+    if not os.path.exists(ruff_package):
```

Comment:
Yes for example, if someone installs ruff in base machine or some other virtualenv instead of the virtualenv where actual Hue is present. Although `make apps` wont do this thing, but I felt it doesn't hurt having this safe check for the command.